subprojects/packagefiles/libtommath/meson.build

```diff
@@ -0,0 +1,380 @@
+libtommath = library(
+  'tommath',
+  sources,
+  install: true,
+  include_directories: inc,
+  version: version,
+  soversion: version_major,
+)
+
+# Install headers
+install_headers(files('tommath.h'))
+
+# Generate pkg-config file
+pkg = import('pkgconfig')
+pkg.generate(
+  libtommath,
+  name: meson.project_name(),
+  description: 'public domain library for manipulating large integer numbers',
+  url: 'https://www.libtom.net/LibTomMath/',
+)
+
+# Create a dependency object for use by other projects
+# This makes the library usable as a Meson subproject
+libtommath_dep = declare_dependency(
+  link_with: libtommath,
+  include_directories: inc,
+)
+
+# Make the dependency available to parent projects
+meson.override_dependency(meson.project_name(), libtommath_dep)
+
+tommath_test_exe = executable(
+  'tommath_test',
+  ['demo/test.c', 'demo/shared.c'],
+  link_with: libtommath,
+  include_directories: inc,
+  install: false,
+  build_by_default: false,
+)
+
+mtest_opponent_exe = executable(
+  'mtest_opponent',
+  ['demo/mtest_opponent.c', 'demo/shared.c'],
+  link_with: libtommath,
+  include_directories: inc,
+  install: false,
+  build_by_default: false,
+)
+
+# Build mtest - reference implementation for differential testing
+# Note: mtest.c includes mpi.c directly, so we only compile mtest.c
+# Have to disable optimization when building mpi.c or the test fails
+# due to bugs in mpi.c
+mtest_exe = executable(
+  'mtest',
+  'mtest/mtest.c',
+  include_directories: include_directories('mtest'),
+  override_options: ['optimization=0'],
+  install: false,
+  build_by_default: false,
+)
+
+# Build and run unit tests if requested
+if get_option('tests')
+  test('tommath_test', tommath_test_exe)
+
+  # Differential test: mtest generates test cases, mtest_opponent validates them
+  # Usage: mtest <iterations> | mtest_opponent
+  test(
+    'mtest_opponent',
+    find_program('sh'),
+    args: ['-c', '"$MTEST" "${1:-1000}" | "$MTEST_OPPONENT"', '--'],  # --: allow iteration count override via --test-args
+    env: environment(
+      {
+        # Use env vars to handle paths with spaces
+        'MTEST': mtest_exe.full_path(),
+        'MTEST_OPPONENT': mtest_opponent_exe.full_path(),
+      },
+    ),
+    depends: [mtest_exe, mtest_opponent_exe],
+    timeout: 30,
+    suite: 'differential',
+  )
+endif
+
+timing_exe = executable(
+  'timing',
+  'demo/timing.c',
+  link_with: libtommath,
+  c_args: ['-DTIMER'],
+  include_directories: inc,
+  install: false,
+  build_by_default: false,
+)
+
+if get_option('benchmarks')
+  # Create logs directory in builddir for benchmark output (timings binary segfaults without it)
+  run_command(
+    'mkdir',
+    '-p',
+    meson.project_build_root() / 'logs',
+    check: true,
+  )
+  benchmark(
+    'benchmarks',
+    timing_exe,
+    timeout: 30,
+  )
+endif
+
+# Documentation target - build PDF manual from LaTeX sources
+if get_option('docs')
```

Comment:
WrapDB wraps don't necessarily reimplement all functionality from the upstream build system.  When a library is being used as a wrap, the parent project won't want the docs, and so we normally wouldn't maintain the additional complexity of `build-manual.py` etc.

On the other hand, if your use case requires this (e.g. because you want to add the Meson build config to the upstream repo), it isn't totally forbidden in WrapDB.